stingray/fourier.py

```diff
@@ -0,0 +1,1286 @@
+    N : int
+        the number of bins in the light curves used in each segment
+    M : int
+        the number of averaged periodograms
+    mean : float
+        the mean flux (geometrical average of the mean fluxes in the two channels)
+
+    """
+    # Initialize stuff
+    unnorm_cross = unnorm_pds1 = unnorm_pds2 = None
+    M = 0
+
+    sum_of_photons1 = sum_of_photons2 = 0
+
+    for flux1, flux2 in zip(flux_iterable1, flux_iterable2):
+        if flux1 is None or flux2 is None:
+            continue
+
+        N = flux1.size
+
+        # Calculate the sum of each light curve, to calculate the mean
+        # This will
+        nph1 = flux1.sum()
+        nph2 = flux2.sum()
+
+        # At the first loop, we define the frequency array and the range of positive
+        # frequency bins (after the first loop, cross will not be None nymore)
+        if unnorm_cross is None:
+            freq = fftfreq(N, dt)
+            fgt0 = positive_fft_bins(N)
+
+        # Calculate the FFTs
+        ft1 = fft(flux1)
+        ft2 = fft(flux2)
+
+        # Calculate the unnormalized cross spectrum
+        unnorm_power = ft1 * ft2.conj()
+
+        # Accumulate the sum to calculate the total mean of the lc
+        sum_of_photons1 += nph1
+        sum_of_photons2 += nph2
+
+        # Take only positive frequencies
+        unnorm_power = unnorm_power[fgt0]
+
+        # Initialize or accumulate final averaged spectrum
+        unnorm_cross = sum_if_not_none_or_initialize(unnorm_cross, unnorm_power)
+
+        M += 1
+
+    # If no valid intervals were found, return only `None`s
+    if unnorm_cross is None:
+        return None
+
+    # Calculate the mean number of photons per chunk
+    Nph1 = sum_of_photons1 / M
+    Nph2 = sum_of_photons2 / M
+    # Calculate the mean number of photons per bin
+    common_mean1 = Nph1 / N
+    common_mean2 = Nph2 / N
+    common_mean = np.sqrt(common_mean1 * common_mean2)
+
+    # Transform the sums into averages
+    unnorm_cross /= M
+
+    # Finally, normalize the cross spectrum (only if not already done on an
+    # interval-to-interval basis)
+    cross = normalize_periodograms(
+        unnorm_cross,
+        dt,
+        N,
+        common_mean,
+        norm=norm,
+        variance=None,
+        power_type="all",
+    )
+
+    # No negative frequencies
+    freq = freq[fgt0]
+
+    results = Table()
+    results["freq"] = freq
+    results["power"] = cross
+    results["unnorm_power"] = unnorm_cross
+    results.meta.update({"n": N, "m": M, "dt": dt,
+                         "norm": norm,
+                         "df": 1 / (dt * N),
+                         "nphots1": Nph1, "nphots2": Nph2,
+                         "mean": common_mean,
+                         "mean1": common_mean1,
+                         "mean2": common_mean2,
+                         "power_type": "all",
+                         "segment_size": dt * N})
+
+    return results
+
+
+def avg_cs_from_iterables(
```

Comment:
So originally we had `Powerspectrum` as a special case of `Crossspectrum`. Is there a specific reason you've split this out into two separate functions? Isn't that going to duplicate a lot of code?

Reply:
Yes, but I don't like much the special case way, which makes the code arguably less legible and generally slower (because of the ifs here and there). We can discuss this, of course

Reply:
What I tried to do here, instead, is making the code accept both arrays of events and series of counts.

Reply:
You're right, the event/light curve compatibility was not for this function